File: 1-two-sum/two-sum.cpp
#include <vector>
#include <unordered_map>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {

        unordered_map<int , int> mp;
        int complement;

        for(int i = 0 ; i < nums.size() ; i++){
            complement = target - nums[i];

            if(mp.count(complement) == 1){
                return {i , mp[complement]};
            }else{
                mp[nums[i]] = i;
            }
        }
        return {};
    }
};

File: 1-two-sum/two-sum.ts


function twoSum(nums: number[], target: number): number[] {


    let map = new Map<number,number>();

    for(let i : number = 0; i < nums.length ; i++){

        let complement : number = target - nums[i];

        if(map.has(complement)){
            return [i , map.get(complement)];
        }

        map.set(nums[i], i);

    }

    return [];
    
};